prody/proteins/compare.py

```diff
@@ -336,8 +336,10 @@ def buildFromChain(self, chain):
             aa = AAMAP.get(res.getResname(), 'X')
             if aa == '-':
                 aa = 'X'
-            if aa == 'X':
+            if aa == 'X' and res.getResname() != 'HOH':
                 LOGGER.warn("no one-letter mapping found for %s " % repr(res))
+            if not res.getResname() in protein_resnames:
+                continue
```

Comment:
Since the function itself can't reliably determine if a residue is really "non-protein" (water, ligand, etc.), or a mutated residue, it's better to (ask the users to) explicitly give the alignment function to the protein part. I think skipping here only suppresses the error can lead to unexpected behaviors.